### ISSUE TITLE
font icons error

### ISSUE THREAD

**Issue by 1m-N00b** [jarun/nnn#853]
# Issues
font error like [that(nnn-nerd and nnn-fonts)](https://imgur.com/a/21V6dig)
Of course I installed [nerd-fonts-complete](https://aur.archlinux.org/packages/nerd-fonts-complete/) and [icons-in-terminal
](https://aur.archlinux.org/packages/icons-in-terminal/)
#  Environment
- Operating System:Arch Linux x86_64
- Desktop Environment:i3
- Terminal Emulator:kitty,(alacritty,tilix,urxvt)
- Shell:fish

**Comment by KlzXS** [comment]
I see youc've installed some packages with the fonts. Have you checked that they are used as the default font in you terminal?

If the terminal is set to use another font you won't be seing icons. Also check the terminals unicode support, but judging by the list you included that shouldn't be a problem.

**Comment by 1m-N00b** [comment]
i added Nerd font mono in kitty,urxvt but didnt appear icons

**Comment by 0xACE** [comment]
@1m-N00b can you get the icons working in the terminal at all? Because it sounds like your fontconfig isn't setup as to your liking

**Comment by 1m-N00b** [comment]
@0xACE 
The devicon is drawn properly in nvim, and the exa --icons command works as well.

**Comment by 1m-N00b** [comment]
I checked with fc-list, and /usr/share/fonts/nerd-fonts-complete is there!

**Comment by 0xACE** [comment]
I don't use fonts with icons on my computers, so I'm not familiar with this topic.

But what did `exa --icons` render? Are you trying to get `NERD font` working or `icons-in-terminal`? How did you compile `nnn`?

Afaik `nnn` just prints some characters, its up to your terminal to render those characters as icons....

**Comment by 1m-N00b** [comment]
Nerd font is used in nvim's devicon and exa.
I also tried to compile with clones(of course both ), and also with nnn-nerd and nnn-icons in aur, but the same error occurred.

**Comment by jarun** [comment]
Adding @snide as well.

**Comment by 0xACE** [comment]
```
echo -e "\ue5ff \ue155"
```
This should print the folder icon in your terminal, what result do you get from this?

**Comment by 1m-N00b** [comment]
Sorry for the late reply.
It appears as ` \ue5ff \ue155` in kiity,urxvt,alacrity terminal

**Comment by KlzXS** [comment]
Sorry for the inconvenience, use `printf "\ue5ff \ue155"` insted. `fish` doesn't like `echo -e` it seems.

If the result is interesting (i.e. not the string you just posted) take a screenshot of that.

**Comment by 1m-N00b** [comment]
[it](https://imgur.com/WCdHjVn.png) seem file icons.

**Comment by KlzXS** [comment]
Weird. I can only imagine it being an ncurses problem.

I tried to get the output you sent first but I have no idea how it comes out as `~E~U`.

**Comment by 0xACE** [comment]
Maybe his implementation of `ncurses` doesn't support `UTF8`/wide characters

**Comment by KlzXS** [comment]
Maybe, but the default arch `ncurses` package does...

**Comment by 1m-N00b** [comment]
how to check it? of course i installed ncurses and im noob so i didnt change some config

**Comment by 0xACE** [comment]
no idea... what output does `nnn & lsof -p $! && kill $_` give ?

**Comment by 1m-N00b** [comment]
I'm using fish shell, and I'm getting an error with `$!`. So I tried with zsh shell.
output:https://imgur.com/a/knYICj2

**Comment by 0xACE** [comment]
Guessing you have the correct ncurses version. Maybe it won't print it because of missing terminfo definition...

What is your `echo $TERM` ? what happens when you run `TERM=xterm-256color nnn`? Have you modified your environment with `tset` or any of its related utilities or any other locale configuration?

Honestly I haven't got these font icons installed on my computer. All I can do is speculate, which isn't very efficient...

from what I can tell `nnn` just calls `addstr()` which inturn calls `addch()`...

what happens when you apply this patch?

```
diff --git a/src/nnn.c b/src/nnn.c
index ed5f537..bb2d331 100644
--- a/src/nnn.c
+++ b/src/nnn.c
@@ -3540,7 +3540,7 @@ static void print_icon(const struct entry *ent, const int attrs)
 		attron(COLOR_PAIR(C_UND + 1 + picon->color));
 	else if (attrs)
 		attron(attrs);
-	addstr(picon->icon);
+	addwstr(picon->icon);
 	if (picon->color)
 		attroff(COLOR_PAIR(C_UND + 1 + picon->color));
 	else if (attrs)
```

**Comment by 1m-N00b** [comment]
Problem solved.

It was a rudimentary mistake.
I'm Japanese, so I set the time and environment language to Japanese, so it shows up in Japanese on my device.
However, Japanese is not suitable for displaying fonts and searching for errors, So I have used 'export LANG=en_US' and set it to English.
In conclusion, it seems that en_US.UTF-8 is correct.
I copied and pasted without thinking from a Japanese technical site,It was a painful experience.


Thank you for your cooperation. I will continue to support the development of nnn.
